ulib/libax/src/cbindings/socket.rs

```diff
@@ -93,6 +92,31 @@ impl Socket {
     }
 }
 
+fn fill_sockaddr(addr: &SocketAddr) -> ctypes::sockaddr {
+    return unsafe {
+        *(&ctypes::sockaddr_in {
+            sin_family: ctypes::AF_INET as u16,
+            sin_port: addr.port.to_be(),
+            sin_addr: ctypes::in_addr {
+                s_addr: u32::from_be_bytes(addr.addr.as_bytes().try_into().unwrap()).to_be(),
+            },
+            sin_zero: [0; 8],
+        } as *const _ as *const ctypes::sockaddr)
+    };
+}
+
+fn load_sockaddr(addr: *const ctypes::sockaddr) -> LinuxResult<SocketAddr> {
+    let mid = unsafe { *(addr as *const ctypes::sockaddr_in) };
```

Comment:
Check `addr` and `addrlen` validation here, and no need to check in other place.